Format data from Instrument controller methods

gau-nernst opened the pull request:
I added codes to format the data returned from Instrument controller methods.
- price: convert to float
- time: convert to `datetime` object, UTC time
- Candles, Order book and Position book: convert to `pd.DataFrame`

I also added some tests for Instrument controller and updated the demo notebook.

I am wondering if we should have a separate low level API (only API wrapper around OANDA API) and a high level API (e.g. get multiple instruments candles in one dataframe).

Please advise me if I have to change or modify anything

gau-nernst committed:
convert Instrument Controller return object to more useable data

use dataframe to make it easier to use
demo in Instrument Demo.py

gau-nernst committed:
Merge remote-tracking branch 'upstream/master'

gau-nernst committed:
format return value from Instrument controller methods

gau-nernst committed:
add tests for Instrument controller

gau-nernst committed:
update instrument demo nb with new formatted methods

gau-nernst committed:
Delete Instrument Demo.py

nimiew commented:
hey, looks good. I am thinking if can have a param called df. When df is true, return dataframe, else just return raw json. Thoughts?

gau-nernst commented:
Hi,
I think that sounds good. Will do that

gau-nernst committed:
add df parameter option for Instrument controller

gau-nernst commented:
Hello,
I have updated the functions with the df parameter, default to True. For order book and position book requests, I also print `time`, `price` and `bucketWidth` when `df=True` because I think those values are quite important for the user to know when a dataframe is returned.
Tests and Instrument controller demo notebook are also updated
